core/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -289,14 +295,20 @@ private void processIndexShardSnapshots(ClusterChangedEvent event) {
         if (newSnapshots.isEmpty() == false) {
             Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
             for (final Map.Entry<Snapshot, Map<ShardId, IndexShardSnapshotStatus>> entry : newSnapshots.entrySet()) {
+                Map<String, IndexId> indicesMap = snapshotIndices.get(entry.getKey());
+                assert indicesMap != null;
                 for (final Map.Entry<ShardId, IndexShardSnapshotStatus> shardEntry : entry.getValue().entrySet()) {
                     final ShardId shardId = shardEntry.getKey();
                     try {
                         final IndexShard indexShard = indicesService.indexServiceSafe(shardId.getIndex()).getShardOrNull(shardId.id());
+                        final IndexId indexId = indicesMap.get(shardId.getIndexName());
+                        if (indexId == null) {
```

Comment:
I combed through the code and I believe it is impossible. I think I will change this to an assert. The indexId is obtained from the SnapshotsInProgress - if the index doesn't exist there, we have other major problems! It should not be possible for this to be `null` but i'll add an assert in any case.